hollow/src/main/java/com/netflix/hollow/api/producer/validation/DuplicateDataDetectionValidator.java

```diff
@@ -25,20 +26,24 @@
 import com.netflix.hollow.core.schema.HollowObjectSchema;
 import com.netflix.hollow.core.schema.HollowSchema;
 import com.netflix.hollow.core.schema.HollowSchema.SchemaType;
+import com.netflix.hollow.core.write.HollowTypeWriteState;
+import com.netflix.hollow.core.write.objectmapper.HollowPrimaryKey;
+import com.netflix.hollow.core.write.objectmapper.HollowTypeMapper;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Objects;
 import java.util.stream.Collectors;
 
 /**
- * This validator uses HollowPrimaryKey definition to find duplicates for a given type.
- * Validator fails if no HollowPrimaryKey is defined for the given type or if given type has duplicate objects.
+ * A validator that fails if two or more records of the same type are present and they have the same primary key.
  * <p>
- * Note that if an object only has primary key fields then the duplicates are deduped by Hollow and will not result
- * in failing this validator. Ex: If Movie type has int id and String name and if primary key is also int id and String
- * name this validator will not fail as the two objects are deduped.
+ * The primary key may be declared on a data model class using the {@link HollowPrimaryKey} annotation, may be
+ * declared more directly using {@link PrimaryKey} with {@link HollowObjectSchema}, or declared explicitly when
+ * instantiating this validator.
  *
- * @author lkanchanapalli {@literal<lavanya65@yahoo.com>}
+ * @apiNote If all a record's fields are associated with a primary key then any duplicates added during the population
```

Comment:
took me a second to realize what you meant here, maybe add something like `(i.e. the object has no fields that are not in the primary key)`

Reply:
```
 * @apiNote If all a record's fields are associated with a primary key (in other words, the record has no
 * fields that are not associated the primary key) then any duplicates added during the population
 * stage will be naturally de-duped resulting in just one record present.  Therefore a validator created for such a
 * record's type will never fail.
```